File: src/test/java/net/haesleinhuepf/clij/ops/examples/MaximumProjection.java

package net.haesleinhuepf.clij.ops.examples;

import java.io.IOException;

import org.junit.Test;

import net.haesleinhuepf.clij.ops.CLIJ_close.CLIJ_close;
import net.haesleinhuepf.clij.ops.CLIJ_maxProjection.CLIJ_maximumZProjection;
import net.haesleinhuepf.clij.ops.CLIJ_pull.CLIJ_pull;
import net.haesleinhuepf.clij.ops.CLIJ_push.CLIJ_push;
import net.imagej.ImageJ;

public class MaximumProjection {

	@Test
	public void run() throws IOException {
		ImageJ ij = new ImageJ();
		Object input = ij.io().open(getClass().getResource("/samples/t1-head.tif").getPath());
		ij.ui().show("input", input);
		// push image to GPU
		Object inputGPU = ij.op().run(CLIJ_push.class, input);
		// max projection
		Object maximumProjected = ij.op().run(CLIJ_maximumZProjection.class,
			inputGPU);
		// show result
		Object result = ij.op().run(CLIJ_pull.class, maximumProjected);
		ij.ui().show("maximum projected", result);

		// cleanup
		ij.op().run(CLIJ_close.class, inputGPU);
		ij.op().run(CLIJ_close.class, maximumProjected);
		ij.op().run(CLIJ_close.class);

	}

	public static void main(String... args) throws IOException {
		MaximumProjection task = new MaximumProjection();
		task.run();
	}
}

File: src/test/resources/jython/maximumProjection.py
# CLIJ example ImageJ Ops jython: maximumProjection.py
#
# This macro shows how maximum projection can be done in the GPU.
#
# Author: Robert Haase, rhaase@mpi-cbg.de
# Author: Deborah Schmidt, frauzufall@mpi-cbg.de
# May 2019
# ---------------------------------------------

#@ IOService io
#@ UIService ui
#@ OpService ops

import os
import inspect
def getResource(file):
    return os.path.dirname(os.path.abspath(inspect.getsourcefile(lambda:0))) + "/" + file


input = io.open(getResource("../samples/t1-head.tif"))
ui.show("input", input)
# push image to GPU
inputGPU = ops.run("CLIJ_push", input)

maximum_projected = ops.run("CLIJ_maximumZProjection", inputGPU)
# show result
result = ops.run("CLIJ_pull", maximum_projected)
ui.show("maximum projection", result)

#cleanup
ops.run("CLIJ_close", inputGPU)
ops.run("CLIJ_close", maximum_projected)

